relayer/crates/starknet-chain-components/src/types/cosmos/client_state.rs

```diff
@@ -101,3 +109,244 @@ impl From<CometClientState> for Any {
+{
+    fn encode_mut(
+        encoding: &Encoding,
+        proof_specs: &InnerSpec,
+        buffer: &mut Encoding::EncodeBuffer,
+    ) -> Result<(), Encoding::Error> {
+        let InnerSpec {
+            child_order,
+            child_size,
+            min_prefix_length,
+            max_prefix_length,
+            empty_child,
+            hash,
+        } = serde_json::to_value(proof_specs)
+            .and_then(serde_json::from_value)
+            .unwrap();
+        //.map_err(|_| Encoding::raise_error("invalid connection end"))?;
+        let child_order = child_order.iter().map(|x| *x as u32).collect();
+        let child_size = child_size as u32;
+        let min_prefix_length = min_prefix_length as u32;
+        let max_prefix_length = max_prefix_length as u32;
+        let hash = hash as u32;
+
+        encoding.encode_mut(
+            &product![
+                child_order,
+                child_size,
+                min_prefix_length,
+                max_prefix_length,
+                empty_child,
+                hash,
+            ],
+            buffer,
+        )?;
+        Ok(())
+    }
+}
+
+#[cgp_provider(MutDecoderComponent)]
+impl<Encoding, Strategy> MutDecoder<Encoding, Strategy, InnerSpec> for EncodeInnerSpec
+where
+    Encoding: CanDecodeMut<Strategy, Product![Vec<u32>, u32, u32, u32, Vec<u8>, u32]>
+        + CanRaiseAsyncError<&'static str>,
+{
+    fn decode_mut<'a>(
+        encoding: &Encoding,
+        buffer: &mut Encoding::DecodeBuffer<'a>,
+    ) -> Result<InnerSpec, Encoding::Error> {
+        let product![
+            child_order,
+            child_size,
+            min_prefix_length,
+            max_prefix_length,
+            empty_child,
+            hash
+        ] = encoding.decode_mut(buffer)?;
+        let child_order = child_order.iter().map(|x| *x as i32).collect();
+        Ok(InnerSpec {
+            child_order,
+            child_size: child_size as i32,
+            min_prefix_length: min_prefix_length as i32,
+            max_prefix_length: max_prefix_length as i32,
+            empty_child,
+            hash: hash as i32,
+        })
+    }
+}
+
+pub struct EncodeProofSpec;
+
+#[cgp_provider(MutEncoderComponent)]
+impl<Encoding, Strategy> MutEncoder<Encoding, Strategy, ProofSpec> for EncodeProofSpec
+where
+    Encoding: CanEncodeMut<Strategy, Product![LeafOp, InnerSpec, u32, u32, bool]>,
+{
+    fn encode_mut(
+        encoding: &Encoding,
+        proof_specs: &ProofSpec,
+        buffer: &mut Encoding::EncodeBuffer,
+    ) -> Result<(), Encoding::Error> {
+        let ProofSpec {
+            leaf_spec,
+            inner_spec,
+            max_depth,
+            min_depth,
+            prehash_key_before_comparison,
+        } = serde_json::to_value(proof_specs)
+            .and_then(serde_json::from_value)
+            .unwrap();
+        //.map_err(|_| Encoding::raise_error("invalid connection end"))?;
```

Comment:
this one too.

Reply:
Done in 421fe1fb11ff66b6b39b9353b73f52ee40da5859